controllers/slicegateway/slicegateway.go

```diff
@@ -1534,3 +1518,30 @@ func (r *SliceGwReconciler) ReconcileIntermediateGatewayDeployments(ctx context.
 
 	return ctrl.Result{}, nil, false
 }
+
+// updateGatewayDeployment updates the gateway client deployments with the relevant updated ports
+// from the workersliceconfig
+func (r *SliceGwReconciler) updateGatewayDeployment(ctx context.Context, c client.Client, g *kubeslicev1beta1.SliceGateway, deployment *appsv1.Deployment, nodePort int) error {
```

Comment:
Can we make this function name as `updateGatewayDeploymentNodePort` because this function is only updating nodePort info in the deployment.